pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -323,7 +327,7 @@ private void processNewSegment(String tableNameWithType, SegmentMetadata segment
         // Cleanup the Zk entry and the segment from the permanent directory if it exists.
         LOGGER.error("Could not move segment {} from table {} to permanent directory", segmentName, tableNameWithType,
             e);
-        _pinotHelixResourceManager.deleteSegment(tableNameWithType, segmentName);
+        deleteSegmentIfNeeded(tableNameWithType, segmentName, segmentUploadStartTime);
         LOGGER.info("Deleted zk entry and segment {} for table {}.", segmentName, tableNameWithType);
```

Comment:
Let's move this log into `deleteSegmentIfNeeded()` when we choose to delete the segment. Same for other places